packages/svelte/src/internal/client/proxy.js

```diff
@@ -167,7 +170,13 @@ export function proxy(value, parent = null, prev) {
 				return v === UNINITIALIZED ? undefined : v;
 			}
 
-			return Reflect.get(target, prop, receiver);
+			v = Reflect.get(target, prop, receiver);
+
+			if (is_proxied_array && array_methods.includes(/** @type {string} */ (prop))) {
+				return batch_onchange(v);
```

Comment:
Can we only do this if we know that we're working with `onchange`. This is going to have a significant hit on performance otherwise.

Reply:
Came here to say the same https://github.com/sveltejs/svelte/pull/15073#issuecomment-2604895320

Reply:
Added an extra check for this.